File: Week-1/SelfDividingNumbers-728/solution.js
// Runtime: 64 ms, faster than 55.31% of JavaScript online submissions for Self Dividing Numbers.
// Memory Usage: 35.5 MB, less than 76.92% of JavaScript online submissions for Self Dividing Numbers.

var selfDividingNumbers = function(left, right) {
  var sdn = [];

  var n = left;

  while (left <= right) {
    let digit = n % 10;

    if (digit == 0 || left % digit !== 0) {
      ++left;
      n = left;
      continue;
    }

    n = parseInt(n / 10);

    if (n == 0) {
      sdn.push(left);
      ++left;
      n = left;
    }
  }

  return sdn;
};

File: Week-1/SelfDividingNumbers-728/solution.cpp
// Runtime: 0 ms, faster than 100.00% of C++ online submissions for Self Dividing Numbers.
// Memory Usage: 8.3 MB, less than 100.00% of C++ online submissions for Self Dividing Numbers.

class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> selfDividingNums;
          
           int n = left;
           while(left <= right){
              
               int digit = n % 10;
                if(digit == 0 || left % digit != 0) {
                    ++left; 
                    n = left; 
                    continue;
                }
                
                n /= 10;
                if(n == 0) {
                    selfDividingNums.push_back(left);
                    ++left;
                    n = left;
                }
          
           }
            
        return selfDividingNums;
    }
};